### PR TITLE
HermesMock with ClassRule and JUnit 5 Extension

### PR DESCRIPTION
**Pull request by cristaloleg**

### CODE REVIEW

File: hermes-mock/build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ dependencies {
     compile group: 'com.jayway.awaitility', name: 'awaitility', version: '1.6.1'
     compile group: 'org.apache.avro', name: 'avro', version: versions.avro
     compile group: 'tech.allegro.schema.json2avro', name: 'converter', version: versions.json2avro
+    compile group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.3.1'
```
Comment (dankraw) [inline hermes-mock/build.gradle]: looks like we have junit 4 and 5 both on classpath?

File: hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockAvroTest.groovy [hunk 2]
```diff
@@ -3,22 +3,30 @@ package pl.allegro.tech.hermes.mock
 import com.fasterxml.jackson.databind.ObjectMapper
 import org.apache.avro.Schema
 import org.apache.avro.reflect.ReflectData
-import org.junit.Rule
+import org.junit.ClassRule
+import org.testng.annotations.BeforeTest
```
Comment (dankraw) [inline hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockAvroTest.groovy]: junit and testng at the same time?

Comment (cristaloleg) [reply hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockAvroTest.groovy]: Good catch, this is an error.

File: hermes-mock/build.gradle (hunk 1 above)
Comment (cristaloleg) [reply hermes-mock/build.gradle]: Jakub has suggested to add support for `Extensions` from JUnit 5.
Objections?

File: hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy [hunk 3]
```diff
@@ -1,15 +1,23 @@
 package pl.allegro.tech.hermes.mock
 
-import org.junit.Rule
+import org.junit.ClassRule
+import org.testng.annotations.BeforeTest
```
Comment (dankraw) [inline hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy]: same here, junit+testng

File: hermes-mock/build.gradle (hunk 1 above)
Comment (dankraw) [reply hermes-mock/build.gradle]: I understand, it's fine.

Comment (dankraw) [reply hermes-mock/build.gradle]: Thanks :)

File: hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy (hunk 3 above)
Comment (dankraw) [reply hermes-mock/src/test/groovy/pl/allegro/tech/hermes/mock/HermesMockTest.groovy]: Oh, it's fixed :)

**Pull request merged**